.github/workflows/generate-and-deploy-with-delete.yml

```diff
@@ -2,31 +2,60 @@
 name: Generate and Sync Files with Delete
 
 on:
-  workflow_dispatch
+  workflow_dispatch:
+    inputs:
+      environment:
+        type: choice
+        description: The environment of the Registry
+        options:
+          - Development
+          - Production
+      bump-versions-branch:
+        type: string
+        description: The branch with the modules and providers to publish (if not specified will use the workflow's ref)
+        required: false
 
 jobs:
   generate-and-sync:
     runs-on: ubuntu-latest
-
-    environment: Production-with-delete
+    environment:
+      name: ${{ inputs.environment }}
 
     steps:
       - name: Checkout Repository
         uses: actions/checkout@v4
+        with:
+          path: main
+
+      - name: Checkout branch with modules and providers
+        if: inputs.bump-versions-branch != ''
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.bump-versions-branch }}
+          path: providers-and-modules
 
       - name: Set up Go
         uses: actions/setup-go@v4
         with:
-          go-version-file: './src/go.mod'
+          go-version-file: './main/src/go.mod'
```

Comment:
I got confused for a moment when I read this... maybe this could be renamed to something like `repo`?

Reply:
agree https://github.com/opentofu/registry/pull/95/commits/bd0e4cb3b9d3e0f2f0931cefe6e207d82b24104d